git/tagging-versions.md

```diff
@@ -0,0 +1,108 @@
+accounting of what was deployed. To that end, I propose the following best
+practice.
+
+<a name="levels"></a>
+## 1. Levels of information
+
+Git already records commits, which include a summary message by the developer
+making the change. This level of information—both the actual files and lines
+changed as well as the summary messages—are the ground-level truth of what
+changed. It should be possible to easily list the commits specific to a version.
+
+The customer typically does not care about this level of information. Instead,
+customers track things at the level of tasks and features. Such features are
+typically tracked in Trello or another project management tool as "cards" (or
+similar). Cards have an associated identifying number. These card numbers should
+be explicitly listed in the information about a deployment using the process
+described below.
+
+<a name="versions"></a>
+## 2. Versions
+
+Each deployment should have a version number associated with it. There is a
+one-to-one relationship between deployments and versions. Each version number is
+in the form `major.minor.patch`, e.g. 2.4.1, where the patch is optional and, if
+not specified, defined to be 0 (so `2.4` is a sort of shorthand for `2.4.0`).
+
+Each regularly scheduled deployment, referred to as a regular deployment in this
+document, should result in a new minor version number. In most cases, the minor
+version number should be incremented. When the project craftsman deems the new
+version to be especially significant, the major version number should be
+incremented and the minor version reset to 0.
+
+(Note: this approach is similar to, but intentionally different
+from, [Semantic Versioning](http://semver.org/), which is a ubiquitous version
+numbering scheme more suitable for libraries. The projects targeted by this best
+practice document are never included as a dependency of another project.)
+
+A "hotfix" occurs when a bug was deployed to production or if, for some other
+reason, a change needs to be deployed outside of the regular schedule. Hotfixes
+should increment the patch number. Each new regular deployment should reset the
+patch number to 0.
+
+<a name="tagging"></a>
+## 3. Tagging versions
+
+Git supports tags, which are names associated with a specific commit. These tags
+may be annotated with a message and sync'd between remotes (e.g. pushed up to
+`origin`).
+
+Each version should be tagged with a tag whose name is `v<version>`, e.g.
+`v2.4.1`. The tag should be annotated with a message that includes the version
+number, a summary if applicable, and a list of the card numbers and card titles
+introduced by this version. The cards should be sorted by number. Here is an
+example message for a tag named `v2.4.1`:
+
+    Version 2.4.1
+
+    Hotfix to fix a typo and fix a permission
+
+    Includes the following cards from the Trello board:
+    - #144 - Correct misspelling of "mispelled"
```

Comment:
Should all the cards be included in all release or just hotfixes?  If all releases, then we may want to add some simple scripting to make that list easier to generate